sympy/physics/mechanics/joint.py

```diff
@@ -0,0 +1,395 @@
+    - ``'_set_linar_velocity'``
+
+    Parameters
+    ==========
+
+    name : string
+        The Joint's name which makes it unique.
+    parent : Body
+        The parent body of joint.
+    child : Body
+        The child body of joint.
+    coordinates: List of dynamicsymbols/dynamicsymbol, optional
+        Coordinates of joint.
+    speeds : List of dynamicsymbols/dynamicsymbol, optional
+        Speeds of joint.
+    parent_joint_pos : Vector, optional
+        Defines the joint's point where the parent will be connected to child.
+        Default value is masscenter of Parent Body.
+    child_joint_pos : Vector, optional
+        Defines the joint's point where the child will be connected to parent.
+        Default value is masscenter of Child Body.
+    parent_axis : Vector, optional
+        Axis of parent frame which would be aligned with child's
+        axis. Default is x axis in parent's frame.
+    child_axis : Vector, optional
+        Axis of child frame which would be aligned with parent's
+        axis. Default is x axis in child's frame.
+
+    Attributes
+    ==========
+
+    name : string
+        The joint's name.
+    parent : Body
+        The joint's parent body.
+    child : Body
+        The joint's child body.
+    coordinates : list
+        List generalized coordinates of the joint.
+    speeds : list
+        List generalized speeds of the joint.
+    parent_joint : Point
+        The joint's point where parent body is connected to the joint.
+    child_joint : Point
+        The joint's point where child body is connected to the joint.
+    parent_axis : Vector
+        The axis of parent frame.
+    child_axis : Vector
+        The axis of child frame.
+
+    """
+
+    def __init__(self, name, parent, child, coordinates=None, speeds=None, parent_joint_pos=None,
+        child_joint_pos=None, parent_axis=None, child_axis=None):
+
+        self.name = name
+        self.parent = parent
+        self.child = child
+
+        self.coordinates = coordinates
+        self.speeds = speeds
+
+        self.parent_axis = parent_axis
+        self.child_axis = child_axis
+
+        self.parent_joint = parent_joint_pos
+        self.child_joint = child_joint_pos
+
+        self._orient_frames()
+        self._set_angular_velocity()
+        self._set_linear_velocity()
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        return self.__str__()
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        if not isinstance(value, str):
+            raise TypeError('Supply a valid name.')
+        self._name=value
+
+    @property
+    def parent(self):
+        """Parent body of Joint."""
+        return self._parent
+
+    @parent.setter
+    def parent(self, value):
+        if not isinstance(value, Body):
+            raise TypeError('Parent must be an instance of Body.')
+        self._parent = value
```

Comment:
You don't define setters if you want the attribute to be immutable. I don't think you want setters, otherwise you have to recreate the joint with the methods in `__init__()`.

Reply:
It gives error without setters.

Reply:
Then look up how to do immutable attributes.

Reply:
Done!